feat(cc-flex-gap): Remove the component and replace it with CSS

florian-sanders-cc opened the pull request:
Fixes #769 

## What does this PR do?

- Removes the `cc-flex-gap` component,
- Replaces it usage with CSS,
- Improves responsiveness of some components.

**Note:**
- [x] This PR does not modify `cc-pricing-*` components and `cc-zone`. It should be merged after the pricing PR.

## Things to check

- every commit,
- compare these components visually with prod:
    - [x] cc-toaster story
    - [x] cc-tcp-redirection
    - [x] cc-invoice-table
    - [x] cc-heptapod-info
    - [x] cc-header-orga
    - [x] cc-header-app
    - [x] cc-header-addon
    - [x] cc-env-var-input
    - [x] cc-env-var-form
    - [x] cc-env-var-create
    - [x] cc-elasticsearch-info
    - [x] cc-addon-credentials
    - [x] cc-addon-linked-apps
    - [x] cc-addon-features

## TODO (Flo)

- [x] Rebase and fixup feedback commits

pdesoyres-cc reviewed (approved):
Well done Flo! I've got nothing to say.

roberttran-cc reviewed (commented):
A very thorough PR, as usual: very nice job!

I have the same feedback as Mathieu on:
- `cc-addon-linked-apps`: `cc-zone` alignment is off with the beginning of the line
- `cc-header-orga`: space is missing between the main name and the badge

I'll also add:
- `cc-header-addon`: breakpoints when content should wrap is broken

florian-sanders-cc commented:
Thanks for your reviews @Galimede @roberttran-cc!

`cc-header-orga`:
Oups, forgot to reset a test I made. I still find the spacing between "hotline" and the badge below weird but it's roughly the same as prod now :+1: 

`cc-addon-linked-apps`:
Fixed :+1: The alignment is more centered than in prod now. I think it's better this way but WDYT?

`cc-header-addon`:
Very good catch @roberttran-cc. I fixed it. There might still be a slight difference but the component is can shrink below 450px compared to the one in prod so I think it's for the best :+1:. 

`cc-env-var-input`: 
You are right @Galimede, I did not see at first but now I do. I'd say we can leave it like that :+1:.

roberttran-cc commented on src/components/cc-zone/cc-zone.js:
```diff
@@ -144,6 +144,7 @@ export class CcZone extends LitElement {
           --lh: 1.5em;
 
           display: flex;
+          align-items: center;
```
**💣 chore:** Actually, to fix `cc-addon-linked-apps` alignment issue, I would not add this rule here but... (1/2)

```suggestion
```

roberttran-cc commented on src/components/cc-addon-linked-apps/cc-addon-linked-apps.js:
```diff
@@ -112,10 +111,10 @@ export class CcAddonLinkedApps extends LitElement {
-          --cc-gap: 0.5em;
-
+          display: flex;
+          flex-wrap: wrap;
```
**💣 chore:** ...but I'd place it there. WDYT? (2/2)

```suggestion
          flex-wrap: wrap;
          align-items: center;
```

Note: actually, it was previously here with `--cc-align-items: center;`.

florian-sanders-cc replied:
You are right! I broke `cc-zone` with my previous change. I'll go for your solution (i'll edit the commit directly instead of creating a new one :+1:)

Thanks @roberttran-cc :rocket:

roberttran-cc reviewed (commented):
`cc-header-orga`, `cc-header-addon`, `cc-env-var-input` : perfect!

`cc-addon-linked-apps`: I made a suggestion on how to fix the issue, tell me what you think about it!

Galimede reviewed (approved):
LGTM, GG! :muscle:
